pkg/util/enrich/enrich.go

```diff
@@ -0,0 +1,61 @@
+package enrich
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/kentik/ktranslate/pkg/eggs/logger"
+	"github.com/kentik/ktranslate/pkg/kt"
+)
+
+type Enricher struct {
+	logger.ContextL
+	url    string
+	client *http.Client
+}
+
+func NewEnricher(url string, log logger.Underlying) (*Enricher, error) {
+	e := Enricher{
+		ContextL: logger.NewContextLFromUnderlying(logger.SContext{S: "Enricher"}, log),
+		url:      url,
+		client:   &http.Client{},
+	}
+
+	e.Infof("Enriching at %s", url)
+	return &e, nil
+}
+
+func (e *Enricher) Enrich(ctx context.Context, msgs []*kt.JCHF) ([]*kt.JCHF, error) {
+
+	target, err := json.Marshal(msgs) // Has to be an array here, no idea why.
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", e.url, bytes.NewBuffer(target))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := e.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
Maybe? I was assuming a non hostile environment, but that's maybe not a good assumption. Could just check Content-Size header and only allow up to a max? / Also handle 0 case?

Reply:
That would work.  I think it's generally safe to assume a non-hostile env but would pretty easy to wreak havok otherwise.  Buffering to disk would be an option but could still fill up if large enough.  The max size sounds like a good middle ground to me.

Reply:
:+1: